### PR TITLE
TRD: Calibration update and additonal macros

### PR DESCRIPTION
**Pull request by f3sch**
I updated the README and added two macros:
- plotVdriftExB.C plots the VDrift and ExB values in the ccdb for a given duration for each chamber
- manualCalibFit.C performs the calibration fits for VDrift and ExB manually

### CODE REVIEW

File: Detectors/TRD/calibration/README.md
Comment (martenole) [inline Detectors/TRD/calibration/README.md]: Actually this is wrong I think (mistake made by me). Can you please change this line to
`o2-trd-global-tracking -b --disable-root-output --enable-trackbased-calib | o2-calibration-trd-vdrift-exb -b --calib-vdexb-calibration '--tf-per-slot 5 --min-entries 50000 --max-delay 90000'`?

**Review by martenole — DISMISSED** [review]
Thanks a lot, this is quite helpful. I think it should also be possible to retrieve for a given run number the GRP object from the CCDB. But then one needs to use not only the time, since multiple runs can be ongoing in parallel. @shahor02 do you know how one can request the GRP object from a macro using the CCDB manager? Is this supported?
Then we could get the run start and stop time from that object and don't need to get it manually from bookkeeping.

**Comment by shahor02** [comment]
> Thanks a lot, this is quite helpful. I think it should also be possible to retrieve for a given run number the GRP object from the CCDB. But then one needs to use not only the time, since multiple runs can be ongoing in parallel. @shahor02 do you know how one can request the GRP object from a macro using the CCDB manager? Is this supported? Then we could get the run start and stop time from that object and don't need to get it manually from bookkeeping.

Hi, one should provide the run number via metadata:
```
  auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
  ...
  std::map<std::string, std::string> md;
  md["runNumber"] = std::to_string(runNumber);
  const auto* grp = ccdbMgr.getSpecific<o2::parameters::GRPECSObject>("GLO/Config/GRPECS", timestamp, md);
```

File: Detectors/TRD/calibration/macros/CMakeLists.txt
Comment (martenole) [inline Detectors/TRD/calibration/macros/CMakeLists.txt:36]: I think you need to add `O2::TRDCalibration` here as well. The CI tests are failing with `/System/Volumes/Data/build/nomad/alloc/627e7c3c-41b1-6bbb-b5b0-3fab21115acf/ci/local/o2/sw/SOURCES/O2/9863/0/Detectors/TRD/calibration/macros/manualCalibFit.C:32:10: fatal error: 'TRDCalibration/CalibratorVdExB.h' file not found`

**Comment by martenole** [comment]
> Hi, one should provide the run number via metadata:
> 
> ```
>   auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
>   ...
>   std::map<std::string, std::string> md;
>   md["runNumber"] = std::to_string(runNumber);
>   const auto* grp = ccdbMgr.getSpecific<o2::parameters::GRPECSObject>("GLO/Config/GRPECS", timestamp, md);
> ```

Thanks a lot Ruben! @f3sch could you add this method such that one does not need to specify time stamps anymore? You can take a look at `DataFormats/Parameters/include/DataFormatsParameters/GRPECSObject.h`. With `grp->getTimeStart()` and `grp->getTimeEnd()` you can get beginning and end time of the run.

**Comment by martenole** [comment]
> Hi, one should provide the run number via metadata:
> 
> ```
>   auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
>   ...
>   std::map<std::string, std::string> md;
>   md["runNumber"] = std::to_string(runNumber);
>   const auto* grp = ccdbMgr.getSpecific<o2::parameters::GRPECSObject>("GLO/Config/GRPECS", timestamp, md);
> ```
Hi again @shahor02 
But this means we still have to provide a timestamp from somewhere between SOR and EOR, right? Or can this also be fetched automatically somehow, e.g. from bookkeeping or is there a helper in O2 already for this?
Edit: what I would like to achieve is that only the run number needs to be provided to automatically get the SOR and EOR times from CCDB

**Comment by f3sch** [comment]
Dont merge yet, waiting for @shahor02 to reply to @martenole.

**Comment by shahor02** [comment]
In the macros, where we can use BasicCCDBManager, one can do:
```
auto& ccdbMgr = o2::ccdb::BasicCCDBManager::instance();
...
auto rr = ccdbMgr.getRunDuration(runNumber);
std::map<std::string, std::string> md;
md["runNumber"] = std::to_string(runNumber);
const auto* grp = ccdbMgr.getSpecific<o2::parameters::GRPECSObject>("GLO/Config/GRPECS", (rr.first+rr.second)/2, md);
```
PS: provided the run was registered in http://alice-ccdb.cern.ch/browse/RCT/Info/RunInformation

**Comment by f3sch** [comment]
Works like a charm, thank you @shahor02.
If CI passes now ready for merge once @martenole approves.

File: Detectors/TRD/calibration/macros/plotVdriftExB.C
Comment (martenole) [inline Detectors/TRD/calibration/macros/plotVdriftExB.C:42]: probably you should set the URL to `ccdb` here

File: Detectors/TRD/calibration/macros/plotVdriftExB.C
Comment (martenole) [inline Detectors/TRD/calibration/macros/plotVdriftExB.C]: Why do you not take 15 minutes (900000) for the `timeOffset`? Also something like `queryInterval` might be more intuitive, but this is up to you

File: Detectors/TRD/calibration/macros/plotVdriftExB.C
Comment (f3sch) [reply Detectors/TRD/calibration/macros/plotVdriftExB.C:42]: Didn't you tell me the GRPECSObject are only alice-ccdb not anywhere
else?
Maybe adding a second parameter for the ccdb for GRPECSObjects is better?

Comment (martenole) [reply Detectors/TRD/calibration/macros/plotVdriftExB.C:42]: Yes. Ah, ok, you are changing the URL below to `ccdb` to get the TRD object from the specified path and only the GRP always from alice-ccdb. This I missed. Then this is fine.

File: Detectors/TRD/calibration/macros/plotVdriftExB.C
Comment (f3sch) [reply Detectors/TRD/calibration/macros/plotVdriftExB.C]: I do not recall the reason for this exactly but I think in the ccdb the offsets are like this.
But changing this is works just as well.
Renaming is probably better.
_There are only two hard things in Computer Science: cache invalidation, naming things and off by one errors._

**Review by martenole — APPROVED** [review]

**Pull request merged**
